main.py

```diff
@@ -0,0 +1,224 @@
+async def process_name(message, state):
+    """
+    Установка имени пользователя.
+    """
+    async with state.proxy() as data:
+        data['name'] = message.text
+
+    await Form.next()
+    await message.reply('Введи дату своего рождения в формате "ДД-ММ-ГГ"')
+
+
+@dp.message_handler(state=Form.birthday)
+async def process_birthday(message, state):
+    """
+    Установка дня рождения.
+    """
+    try:
+        birthday = dt.strptime(message.text, '%d-%m-%y')
+    except ValueError:
+        await bot.send_message(message.chat.id, 'Введите дату в правильном формате')
+        return
+    else:
+        async with state.proxy() as data:
+            data['birthday'] = birthday
+            data['congratulations'] = False
+    await Form.next()
+    await message.reply('Введи любое сообщение что бы начать!')
+
+
+@dp.message_handler(state=Form.yes_or_not)
+async def message_loop(message, state):
+    """
+    Основной цикл бота.
+    """
+    await message.reply('С этого момента ты будешь получать сообщения!')
+
+    redis_connection = await aioredis.create_connection("redis://localhost")
+    redis = await aioredis.Redis(pool_or_conn=redis_connection)
+    await redis.sadd('users:set', message.chat.id)
+
+    async with state.proxy() as data:
+        birthday = data['birthday']
+        data['answer'] = True
+
+    current_state = await state.get_state()
+
+    while current_state is not None:
+        async with state.proxy() as data:
+            if data.get('answer') is None:
+                await redis.srem('users:set', message.chat.id)
+                break
+        chat_id = message.chat.id
+        current_state = await state.get_state()
+        date_today = dt.now()
+        time_now = date_today.time()
+        table = f'user:list:{chat_id}'
+        messages = await redis.hgetall(table, encoding='utf-8')
+        async with state.proxy() as data:
+            congratulations = data['congratulations']
+            date_today_str = date_today.strftime('%d-%m')
+            birthday_str = birthday.strftime('%d-%m')
+            if date_today_str == birthday_str:
+                await happy_birthday(state, congratulations, chat_id)
+            elif congratulations:
+                data['congratulations'] = False
+        if time_now < constants.MORNING['hour_start']:
+            if messages:
+                for message in messages.keys():
+                    await redis.hdel(table, message)
+            await asyncio.sleep(1200)
+            continue
+
+        if not messages:
+            await set_messages(table, redis)
+        time_now_str = time_now.strftime('%H:%m')
+        message_now = await redis.hget(table, time_now_str)
+        if message_now:
+            await send_with_typing(
+                bot_client=bot,
+                chat_id=chat_id,
+                message=message_now
+            )
+            await redis.hdel(table, time_now_str)
+        await asyncio.sleep(30)
```

Comment:
The function is too big. You might better decompose it into less complex and smaller functions.